lib/policyEvaluator/utils/conditions.js

```diff
@@ -0,0 +1,407 @@
+'use strict'; // eslint-disable-line strict
+
+
+const ip = require('ip');
+
+const handleWildcards = require('./wildcards.js').handleWildcards;
+const checkArnMatch = require('./checkArnMatch.js');
+const conditions = {};
+
+/**
+ * findConditionKey finds the value of a condition key based on requestContext
+ * @param {string} key - condition key name
+ * @param {object} requestContext - info sent with request
+ * @return {string} condition key value
+ */
+conditions.findConditionKey = (key, requestContext) => {
+    // TODO: Consider combining with findVariable function if no benefit
+    // to keeping separate
+    const map = new Map();
```

Comment:
Instead of doing all those separate `.set()` you could use the `new Map(iterator)`